python/examples/flight/server.py

```diff
@@ -46,6 +68,26 @@ def do_put(self, descriptor, reader):
         self.flights[key] = reader.read_all()
         print(self.flights[key])
 
+    def do_get(self, ticket):
+        print(ticket.ticket)
+        key = ast.literal_eval(ticket.ticket.decode())
+        if key not in self.flights:
+            return None
+        return pyarrow.flight.RecordBatchStream(self.flights[key])
+
+    def list_actions(self):
+        return [
+            ("clear", "Clear the stored flights."),
+            ("shutdown", "Shut down this server."),
+        ]
+
+    def do_action(self, action):
+        if action.type == "clear":
+            raise NotImplementedError(
+                "{} is not implemented.".format(action.type))
+        else:
+            yield pyarrow.flight.Result(pyarrow.py_buffer(b'Shutdown!'))
```

Comment:
Hmm... I suppose this method should actually shut down the server, instead?

Reply:
I've changed this, though there is a caveat: you have to call `shutdown` on a background thread, as otherwise it blocks waiting for requests to finish, and consequently your `DoAction` request would block on itself.